docs/guide/api-vite-environment.md

```diff
@@ -782,15 +789,17 @@ const runner = new ModuleRunner(
 import { BroadcastChannel } from 'node:worker_threads'
 import { createServer, RemoteEnvironmentTransport, DevEnvironment } from 'vite'
 
-function createWorkerEnvironment(name, config) {
+function createWorkerEnvironment(name, config, context) {
   const worker = new Worker('./worker.js')
   return new DevEnvironment(name, config, {
+    hot: /* custom hot channel */,
     runner: {
       transport: new RemoteEnvironmentTransport({
         send: (data) => worker.postMessage(data),
         onMessage: (listener) => worker.on('message', listener),
       }),
     },
+    watcher: context.watcher
```

Comment:
For now, I moved to `watcher: false | FSWatcher` to force the user to pass the watcher forward. I tried doing this during `init()` but it doesn't feel right. Maybe it is ok for `createEnvironment` to be a bit more complex to implement given that only framework and runtime maintainers will be doing this?

Reply:
The `watcher` is only used in `transformRequest`. Maybe we can find a way to remove it from there somehow so it's not required in the environment at all?

Reply:
I just re-checked and we also need the watcher to implement the plugin context `addWatchFile()`. And I found a bug, as we weren't passing the watcher when creating the plugin container for each environment. I don't think we can remove this. I moved it to `environment.init({ watcher })` at https://github.com/vitejs/vite/pull/17791/commits/569f71486c7098b0f6a998fda8148a4099c559af. We can decide if we want `environment.watcher` or we leave it as internal as it is used now `environment.pluginContainer.watcher`. Does this look better?